### ISSUE TITLE
Add option to disable single instance mode

### ISSUE THREAD

**Issue by SuperDuperDeou** [Vladimir-csp/xdg-terminal-exec#59]
As time goes on, more and more terminal support for single instance mode.

With single instance, only one process of the terminal is ever up, if another terminal process is start, it exits immediately, and the old process spawns the new window.

This is usually desirable: it saves on load times and memory usage, but if the terminal is being started as part of a script, the script will misinterpret the process exiting immediately as the task being completed, thus the need for a way to force the terminal to not run in single instance mode, if needed.

`X-TerminalArgForceNewInstace` should contained the argument to give to the terminal to force the new window to be in a new instance. If this is not set or set but empty, it is assumed that the terminal does not support single instance mode (each terminal is its own instance).

**Comment by SuperDuperDeou** [comment]
For the purposes of this flag, we can roughly group terminals in 3 types:

- **Xterm-like:** Terminals that do not support single instance mode. They're assumed to be synchronous and this does absolutely nothing.
- **Kitty-like:** Terminals that do not support single instance mode, and the synchronous flag still opens the terminal window with the main process, leaving the calling process as a lightweight dummy process. While adding `X-TerminalArgSynchronous` is the proper way of doing this, yes, you can get away baking the synchronous arg into `Exec` and this change wouldn't be necessary.
- **Wezterm-like:** Terminals that support single instance mode, and the synchronous flag opens a *full new instance of a terminal process*. With these terminals, baking the synchronous flag into `Exec=` is exactly the same as running it with *without* single instance mode. You generally want to take advantage of the single instance capabilities: it saves on system resources and the terminal window opens more quickly, but there are times where you really want the terminal window to be synchronous, and you'd rather take the resource hog only when you really need to, hence this proposal.

**Comment by arrowd** [comment]
> This can be achieved by still opening the window in single instance mode, but keeping the process launched by the script alive until the window is closed (ex. Kitty's --wait-for-single-instance-window-close) 

I didn't get that part. Let's say I have a Kitty window open with some work I'm doing. Then I run a script that uses this flag and it causes Kitty window to create a new "tab" for the command from the script. Now the script won't continue until I close both "tabs"? I have to interrupt my work in the first "tab"?

**Comment by SuperDuperDeou** [comment]
> > This can be achieved by still opening the window in single instance mode, but keeping the process launched by the script alive until the window is closed (ex. Kitty's --wait-for-single-instance-window-close)
> 
> I didn't get that part. Let's say I have a Kitty window open with some work I'm doing. Then I run a script that uses this flag and it causes Kitty window to create a new "tab" for the command from the script. Now the script won't continue until I close both "tabs"? I have to interrupt my work in the first "tab"?

No, the script will wait just for *the one window the script opened*, not for every single Kitty window. Without that flag, the script will keep going as soon as the window is *opened*, which is not always the desired behavior.
